.github/workflows/scheduled.yml

```diff
@@ -71,46 +70,161 @@ env:
-          key: ccache-fuzzer
+          key: ccache-fuzzer-centos
+
+      - name: Fix git permissions
+        working-directory: ${{ github.workspace }}
+          # Usually actions/checkout does this but as we run in a container
+          # it doesn't work
+        run: |
+          git config --global --add safe.directory /__w/velox/velox/velox
+          git config --global --add safe.directory /__w/velox/velox/velox_main
+
+      - name: Ensure Stash Dirs Exists
+        working-directory: ${{ github.workspace }}
+        run: |
+          mkdir -p '${{ env.CCACHE_DIR }}'
+          mkdir -p /tmp/signatures
+          # this can be removed after #8917
+          dnf install -q -y python39-devel
+
+      - name: Checkout Main
+        if: ${{ steps.get-sig.outputs.stash-hit != 'true' }}
+        uses: actions/checkout@v4
+        with:
+          # hardcode ref without broken pr
+          ref: 'main'
+          path: velox_main
+
+      - name: Build PyVelox
+        if: ${{ steps.get-sig.outputs.stash-hit != 'true' }}
+        env:
+          CCACHE_BASEDIR: "/__w/velox/velox/velox_main"
+        working-directory: velox_main
+        run: |
+          python3 -m venv .venv
+          source .venv/bin/activate
+
+          make python-build
+
+      - name: Create Baseline Signatures
+        if: ${{ steps.get-sig.outputs.stash-hit != 'true' }}
+        working-directory: velox_main
+        run: |
+          source .venv/bin/activate
+          python3 -m pip install deepdiff
+          python3 scripts/signature.py export --spark /tmp/signatures/spark_signatures_main.json
+          python3 scripts/signature.py export --presto /tmp/signatures/presto_signatures_main.json
+
+      - name: Save Function Signature Stash
+        if: ${{ steps.get-sig.outputs.stash-hit != 'true' }}
+        uses: assignUser/stash/save@v1
+        with:
+          path: /tmp/signatures
+          key: function-signatures
 
-      - name: "Checkout Repo"
+      - name: Checkout Contender
         uses: actions/checkout@v4
         with:
           path: velox
           submodules: 'recursive'
           ref: "${{ inputs.ref }}"
 
-      - name: "Install dependencies"
+      - name: Zero Ccache Statistics
         run: |
-          cd  velox
-          source ./scripts/setup-ubuntu.sh
-          ccache -vsz
+          ccache -sz
 
       - name: Build
+        env:
+          CCACHE_BASEDIR: "/__w/velox/velox/velox"
+          EXTRA_CMAKE_FLAGS: "-DVELOX_ENABLE_ARROW=ON -DVELOX_BUILD_PYTHON_PACKAGE=ON ${{ inputs.extraCMakeFlags }}"
         run: |
-          cd velox
-          make debug NUM_THREADS="${{ inputs.numThreads || 8 }}" MAX_HIGH_MEM_JOBS="${{ inputs.maxHighMemJobs || 8 }}" MAX_LINK_JOBS="${{ inputs.maxLinkJobs || 4 }}" EXTRA_CMAKE_FLAGS="-DVELOX_ENABLE_ARROW=ON ${{ inputs.extraCMakeFlags }}"
+          EXTRA_CMAKE_FLAGS="-DPYTHON_EXECUTABLE=$(python3 -c "import sys; print(sys.executable)") $EXTRA_CMAKE_FLAGS"
```

Comment:
this should be equivalent to `which python3` right ?

Reply:
Yes that should also work, especially now with the image being used only having the one pyhon version